### ISSUE TITLE
Avoid creating analysis records until `--commit`

### ISSUE THREAD

**Issue by funnell** [papaemmelab/isabl_cli#55]
A problem I've encountered is misspecifying application filters can result in a bunch of unwanted analysis records.

What do you think of something like:
1. no `--commit`
* check tuples for validity
* report how many new analyses would be created, how many unprotected would be run, how many are invalid
3. with `--commit`
* create new analysis records
* run the analyses

**Comment by juanesarango** [comment]
Yes, I like this idea. Here we can create some logic to skip analyses creations:

https://github.com/papaemmelab/isabl_cli/blob/78c6c44bd3d483c843895c0d3bfd92585f33cc93/isabl_cli/app.py#L1478-L1504

And just show a `dry-run` info of how many analyses exist and how many will be created. 
Are you open to work in this PR?

**Comment by funnell** [comment]
Yeah I'm open to working on this
